### ISSUE TITLE
Fix helpers A11yNotice strange jest-styled-components behavior

### ISSUE THREAD

**Issue by igorschoester** [Yoast/javascript#237]
## Explanation

First, A11yNotice is a styled component, not a test helper. It is located in the `src/tests` folder and should be just in `src`.

Second, we got unexpected snapshot changes for the usage of A11yNotice in the plugin. See this commit: https://github.com/Yoast/wordpress-seo/pull/12705/commits/d9021c48e29e28e7a01cc9c50ba6040bfc90d4aa
`jest-styled-components` makes it so the styles of styled components are output in the snapshots. This is indeed how the snapshots in the helper look like (notice the `.cX` classes).
However, in the plugin commit above, these are instead removed.

Please make it so that in the plugin the tests also include the `jest-styled-components`'s way of adding `.cX` classes in the snapshots.

**Comment by Xyfi** [comment]
I think it has to do with the fact that they have been moved to separate packages.

My hypothesis is that `jest-styled-components` behaves differently in regard to relative imports than in regard to `@yoast/*` imports.

**Comment by igorschoester** [comment]
Good point, maybe it is fine. Still would be nice to be confirmed though.
And the moving out of the test folder still remains.

**Comment by igorschoester** [comment]
As discussed with Anton, repurposing this PR for only the first point (moving from the test folder).

The second (no A11yNotice styled components output in snapshots) is no longer important enough. It might be expected behavior as Alexander says above. Or it may even be fixed within `jest-styled-components` itself if it is a `styled-components` v4 issue.
